--- tsc2007_core.c	2024-09-17 08:16:45.187562763 +0200
+++ tsc2007_core_fixed.c	2024-09-17 08:15:55.523726897 +0200
@@ -333,17 +333,12 @@
 
 	if (!device_property_read_u32(dev, "ti,x-plate-ohms", &val32)) {
 		ts->x_plate_ohms = val32;
+		ts->y_plate_ohms = val32;
 	} else {
 		dev_err(dev, "Missing ti,x-plate-ohms device property\n");
 		return -EINVAL;
 	}
 
-	if (!device_property_read_u32(dev, "ti,y-plate-ohms", &val32)) {
-		ts->y_plate_ohms = val32;
-	} else {
-		dev_err(dev, "Missing ti,y-plate-ohms device property\n");
-		return -EINVAL;
-	}
 
 	ts->gpiod = devm_gpiod_get_optional(dev, NULL, GPIOD_IN);
 	if (IS_ERR(ts->gpiod))
